argus/introspection/cloud/windows.py

```diff
@@ -68,6 +68,27 @@ def _escape_path(path):
     return path
 
 
+def _getip_v4_v6(ips_as_string, single=True, v6_count=2):
+    """Returns viable v4 and v6 IPs from a space separated string.
+
+    :param single: returns a relevant IP (not all) if True
+    :param v6_count: minimum number of v6 IPs to consider the selection valid
+    """
+    ips = ips_as_string.split(" ")
+    ips_v4, ips_v6 = [], []
+    for ip in ips:
+        if not ip:
+            continue
+        if "." in ip and ":" not in ip:
+            ips_v4.append(ip)
+        else:
+            ips_v6.append(ip)
+    if not single:
+        return ips_v4, ips_v6
+    return (ips_v4[0] if ips_v4 else None,
```

Comment:
I dislike this solution. You control the output through two separate flags and the result is something unmanageable. Here's something better:

1. use something such as the following code in network_details.ps1. Basically, you're annotating each line with whatever field you're wanting.

       echo $sep
       [string]::Format("MAC {0}", $nic.MACAddress -join " ")
       [string]::Format("ip {0}", ($nic.IPAddress -join " "))
       [string]::Format("DefaultIpGateway {0}", ($nic.DefaultIPGateway -join " "))
       [string]::Format("Subnet {0}", ($nic.IPSubnet -join " "))

2. parse this result and use a collections.namedtuple for each component, the result being something like this:

result = network_details.mac.ipv4
            network_details.mac.ipv6
            network_details.ip.ipv4
            network_details.ip.ipv6
            etc.

Which means that you'll be interested in parsing each line. Split it by space, if it has two entries, the first ip v4, the second is ip v6. How does this sound?